oauth2client/service_account.py

```diff
@@ -461,3 +465,94 @@ def create_delegated(self, sub):
         result._private_key_pkcs12 = self._private_key_pkcs12
         result._private_key_password = self._private_key_password
         return result
+
+
+def _datetime_to_secs(utc_time):
+   # time_delta.total_seconds() not supported in Python 2.6
+   epoch = datetime.datetime(1970, 1, 1)
+   time_delta = utc_time - epoch
+   return time_delta.days * 86400 + time_delta.seconds
+
+
+class JWTAccessCredentials(GoogleCredentials):
```

Comment:
Well, that's exactly what I'm waffling on and trying to understand. Is it part of the required developer experience of the feature being implemented that code and developers outside the `oauth2client` library be aware of a JWT access credentials being a kind of thing that is different from a `GoogleCredentials`? Do we support users being able to look at them and examine them and discover "oh, this first one is a `GoogleCredentials` and I can do some things with it but this second one is a `JWTAccessCredentials` and I can do everything with it that I could with the first plus more"?

Or is the situation more like "I need a `GoogleCredentials`, I have a `ServiceAccountCredentials` and some `additional_claims`; ooh look here's a function that I can call that will give me what I need from what I have!"?

To directly answer the question "Why hide the definition of a class [of which] [one] intend[s] people to have instances?": it's appropriate when one is (1) writing a library for a large user base to be supported for a long period of time, and a new public class would increase the support load going forward and (2) the class being proposed doesn't extend the public API of the type according to which its users would use it.

Is that the case here? Can users do anything with a `JWTAccessCredentials` that they couldn't do with a `GoogleCredentials`? Does `JWTAccessCredentials` afford any more methods than `GoogleCredentials`?